File: we_4_2_the_sagnac_phase_of_a_fiber.py
# Integrated Navigation Systems — Chapter 4: Inertial Sensors and Error Modeling
# Worked Example 4.2 — The Sagnac phase of a fiber-optic gyroscope
# Companion code (educational reference implementation; see the book for context).

import numpy as np
L,D,lam,c,Om=200,0.08,1.55e-6,3e8,7.2921150e-5
dphi=2*np.pi*L*D*Om/(lam*c); print(f'dphi={dphi:.3e} rad')   # 1.577e-05

File: we_4_2_the_sagnac_phase_of_a_fiber.m
% Integrated Navigation Systems — Chapter 4: Inertial Sensors and Error Modeling
% Worked Example 4.2 — The Sagnac phase of a fiber-optic gyroscope
% Companion code (educational reference implementation; see the book for context).

L=200; D=0.08; lam=1.55e-6; c=3e8; Om=7.2921150e-5;
dphi=2*pi*L*D*Om/(lam*c); fprintf('dphi=%.3e rad\n',dphi);   % 1.577e-05
